llvm/include/llvm/ADT/GenericCycleInfo.h

```diff
@@ -124,7 +141,12 @@ template <typename ContextT> class GenericCycle {
   bool contains(const GenericCycle *C) const;
 
   const GenericCycle *getParentCycle() const { return ParentCycle; }
-  GenericCycle *getParentCycle() { return ParentCycle; }
+
+  GenericCycle *getParentCycle() {
+    clearCache();
+    return ParentCycle;
+  }
```

Comment:
Clearing the cache in a getter (and only the non-const one) seems dubious

Reply:
Yeah, clearly there is no change in state here and we don't need to clear the cache. At least this one is obvious enough. We could later revisit other non-const functions too, maybe as a future enhancement.